webextension-polyfill updates (#44951)
mdn/content: files/en-us/mozilla/add-ons/webextensions/build_a_cross_browser_extension/index.md

Reviewer: ```suggestion
```

There are two API namespaces in use among the main browsers:

> - `browser.*`, the standard for the extensions API used by Firefox Safari, Chrome (from 148), Opera (from 121), and Edge (from 136).

Reviewer: ```suggestion
- `browser.*`, the standard for the extensions API. Originally used by Firefox and Safari, supported by Chrome [in mid 2026](https://developer.chrome.com/docs/extensions/develop/concepts/browser-namespace) (e.g. Chrome 148, Opera 121, Edge 136, or 4 versions later for extensions using `devtools_page`).
```

Rephrased again for the same reason as https://github.com/mozilla/extension-workshop/pull/2401#discussion_r3700209951

And then I appended precise version numbers anyway for those who really care about precise version numbers. Not sure if we should include that detail here, or just refer to https://github.com/mdn/content/pull/44959 instead (expanding the section as needed).

I lost track of where we put what info due to the multiple PRs and my delay in reviewing some of them.

Firefox also supports callbacks for the APIs that support the `chrome.*` namespace. However, using promises is recommended. Promises greatly simplify asynchronous event handling, particularly when you need to chain events. If you want to use the `browser.*` namespace with promises and target Chrome 147 or earlier, you need to use a polyfill or similar.

> [!NOTE]
> If you're unfamiliar with the differences between these two methods, look at [Getting to know asynchronous JavaScript: Callbacks, Promises and Async/Await](https://medium.com/codebuddies/getting-to-know-asynchronous-javascript-callbacks-promises-and-async-await-17e0673281ee) or the MDN [Using promises](/en-US/docs/Web/JavaScript/Guide/Using_promises) page.

#### The WebExtension browser API Polyfill

Starting with Chrome 148, Chrome supports the `browser` namespace, except in extensions that include a DevTools page. This limitation was removed in Chrome 152 (see [Chrome bug 500769389](https://crbug.com/500769389)), and the `browser` namespace became available to all Chrome extensions.

> To take advantage of the `browser.*` namespace and promises in earlier versions, use the [WebExtension browser API Polyfill](https://github.com/mozilla/webextension-polyfill/).This polyfill addresses the API namespace and asynchronous event handling across Firefox, Chrome, Opera, and Edge.

Reviewer: Add space after period.

```suggestion
To take advantage of the `browser.*` namespace and promises in earlier versions, use the [WebExtension browser API Polyfill](https://github.com/mozilla/webextension-polyfill/). This polyfill addresses the API namespace and asynchronous event handling across Firefox, Chrome, Opera, and Edge.
```

> [!NOTE]

> > In Chrome 148 or later, the polyfill is no-op.

Reviewer: Suggesting 152 here because 148 is technically inaccurate, for extensions with devtools_page.

```suggestion
> In Chrome 152 or later, the polyfill is no-op.
```
